streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Suppressed;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.Change;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.apache.kafka.streams.processor.MockProcessorContext;
+import org.apache.kafka.test.MockInternalProcessorContext;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Random;
+
+import static java.time.Duration.ZERO;
+import static org.apache.kafka.streams.kstream.Suppressed.BufferConfig.unbounded;
+import static org.apache.kafka.streams.kstream.Suppressed.IntermediateSuppression.withEmitAfter;
+import static org.apache.kafka.streams.kstream.Suppressed.emitFinalResultsOnly;
+import static org.apache.kafka.streams.kstream.Suppressed.intermediateEvents;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+@SuppressWarnings("PointlessArithmeticExpression")
+public class KTableSuppressProcessorTest {
+    // Using a static seed gives us deterministic (aka repeatable) tests while documenting which test data are arbitrary
+    private static final Random RANDOM = new Random(42);
```

Comment:
I am fine with this; however, it might be better to take `System.currentTimeMillis()` as seed and LOG the seed -- if it fails, we can reproduced using the logged seed?

Also: it's multiple test methods and I don't think that the execution order is defined. Thus, we should initialize the with known seed, for each test, ie, via `@Before`

Reply:
What you are describing is what I had initially. It seemed a little too fancy, though. I think if we like this approach, we should consider bringing in a real fuzzing framework instead of hand-rolling it.

about `@Before`, this is a good point. I'll do it.

Reply:
Ack. Fine with me to not go too fancy. However, for this case I don't see why we need `Random` at all, and not just hardcode some values for window-start-time etc -- if we seed with `42`, we will get some (unknown) but fixed values anyway -- so what do we gain to use `Random` ? We can just put some fixed values into the code directly.